// --------------------------------------------------------------------------------
// SPDX-FileCopyrightText: 2024-2025 Martin Jan KÃ¶hler and Harald Pretl
// Johannes Kepler University, Institute for Integrated Circuits.
//
// This file is part of KPEX 
// (see https://github.com/martinjankoehler/klayout-pex).
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.
// SPDX-License-Identifier: GPL-3.0-or-later
// --------------------------------------------------------------------------------
syntax = "proto3";

package kpex.klayout;

import "kpex/layout/layer_ref.proto";

message RExtractorTech {
    enum Algorithm {
        ALGORTTHM_UNSPECIFIED = 0;
        ALGORITHM_SQUARE_COUNTING = 1;
        ALGORITHM_TESSELATION = 2;
    }

    message Conductor {
        kpex.layout.LayerRef layer = 10;
        Algorithm algorithm = 20;
        double triangulation_min_b = 30;
        double triangulation_max_area = 31;
        double resistance = 40;
    }

    message Via {
        kpex.layout.LayerRef layer = 10;
        kpex.layout.LayerRef bottom_conductor = 20;
        kpex.layout.LayerRef top_conductor = 30;

        double resistance = 40;
        double merge_distance = 50;
    }

    repeated Conductor conductors = 10;
    repeated Via vias = 20;

    bool skip_simplify = 40;
}
